.github/workflows/linting.yml

```diff
@@ -0,0 +1,19 @@
+name: Linting
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: pip install -e .[dev]
```

Comment:
Do we need this, if `make lint` already sets up the environment and uses it?

Reply:
No, this is not needed. I wanted to make some small changes to the makefile and then rewrite the workflows. Currently each time when calling `make lint` the virtual environment gets created a second time. This is not always necessary. Therefore I thought about checking if the virtual environment already exists and only creating it if there is no one available.